src/workerd/api/js-rpc-target.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2017-2023 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+#pragma once
+
+#include <workerd/api/basics.h>
+#include <workerd/io/worker-interface.capnp.h>
+
+namespace workerd::api {
+
+// The capability that lets us call remote methods over RPC.
+// The client capability is dropped after each callRequest().
+class JsRpcTargetImpl final : public rpc::JsRpcTarget::Server {
+public:
+  JsRpcTargetImpl(
+      kj::Own<kj::PromiseFulfiller<void>> callFulfiller,
+      IoContext& ctx,
+      kj::Maybe<kj::StringPtr> entrypointName)
+      : callFulfiller(kj::mv(callFulfiller)), ctx(ctx), entrypointName(entrypointName) {}
+
+  ~JsRpcTargetImpl() {
+    // The `GetJsRpcTarget` is waiting on the promise associated with callFulfiller; since we've
+    // used the capability, we can fulfill the promise.
+    callFulfiller->fulfill();
```

Comment:
It might be more appropriate to fulfill this after the first call returns, on the assumption that there's only one call invoked. Otherwise you're waiting for a network round trip: the return value has to reach the client before the client will then drop its JsRpcTarget capability.

Of course, this fulfiller only makes sense at all for the top-level object. In v2, when we have the ability to introduce additional objects during the session, they won't have anything like `callFulfiller`...